tests/lib/rules/no-string-refs.js

```diff
@@ -140,10 +145,28 @@ ruleTester.run('no-refs', rule, {
         });
       `,
       options: [{ noTemplateLiterals: true }],
+      settings: { react: { version: '18.2.0' } },
       errors: [
         { messageId: 'thisRefsDeprecated' },
         { messageId: 'stringInRefDeprecated' },
       ],
     },
+    {
+      code: `
+        var Hello = createReactClass({
+          componentDidMount: function() {
+          var component = this.refs.hello;
+          },
+          render: function() {
+            return <div ref={\`hello\${index}\`}>Hello {this.props.name}</div>;
+          }
+        });
+      `,
+      options: [{ noTemplateLiterals: true }],
+      settings: { react: { version: '18.3.0' } },
+      errors: [
+        { messageId: 'stringInRefDeprecated' },
+      ],
+    },
```

Comment:
shouldn't this be valid if the version is 18.3.0?

Reply:
Sorry for the confusion. It is invalid because of the string ref but not the `this.refs` usage. This is identical to the test case above except there is only one error instead of two.

Reply:
aha, thanks

Reply:
I just pushed an update to add a valid test case based one of the invalid test cases but only changing the react version. I hope that helps.